src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl service = new RegistrationServiceImpl();
+    private StorageDaoImpl storage;
+
+    @BeforeEach
+    void setUp() {
+        storage = new StorageDaoImpl();
+        service = new RegistrationServiceImpl();
```

Comment:
The storage instance should be initialized directly without the setUp method, or it should be cleared before each test to ensure test independence.